lib/parse-dir.js

```diff
@@ -1,168 +1,183 @@
-(function() {
-  'use strict';
-
-  var glob = require('glob');
-  var async = require('async');
-  var fs = require('fs');
-  var path = require('path');
-  var noop = function() {};
-
-  // Parsers
-  var jsonParser = require('./parsers/json');
-  var yamlParser = require('./parsers/yaml');
-  var jsParser = require('./parsers/javascript');
-
-  var ParseDir = function() {
-    var that = this;
-
-    var sync = function (pattern) {
-      that.data = [];
-      that.__assync = true;
-
-      var files = glob.sync(pattern);
-
-      if (files) {
-        try {
-          for (var i = 0, len = files.length; i < len; i++) {
-            var file = files[i];
-
-            that.parseFile.call(that, file, noop);
-          }
-          return that.data;
-        } catch (e) {
-          return e;
-        }
-      } else {
-        return [];
+'use strict';
+
+const glob = require('glob');
+const async = require('async');
+const fs = require('fs');
+const path = require('path');
+const noop = function() {};
+
+// Parsers
+const jsonParser = require('./parsers/json');
+const yamlParser = require('./parsers/yaml');
+const jsParser = require('./parsers/javascript');
+
+// used by ParseDir to process and parse individual files:
+class FileProcessor {
+  constructor(options) {
+    this.data = [];
+    this.__async = options.async !== undefined ? options.aync : true;
+  }
+
+  __determineIfDirectory (done, results) {
+    const onLsStat = (err, stats) => {
+      let error = null;
+      if (err || !stats) {
+        error = err || true;
       }
-    };
-
-    var handleAsync = function(pattern, callback) {
-      that.data = [];
-      that.__assync = typeof callback === 'function';
-
-      if (that.__assync) {
-        glob(pattern, function(err, files) {
-          if (err) {
-            callback(err);
-            return;
-          }
-
-          async.each(files, that.parseFile.bind(that), function(err) {
-            callback(err, that.data);
-          });
-        });
-
-        return;
+      if (!error) {
+        return done(null, stats.isDirectory());
       }
-
-      return sync(pattern);
+      return done(error);
     };
-    handleAsync.sync = sync.bind(that);
-
-    return handleAsync;
-  };
-
-  ParseDir.prototype.parseFile = function(file, callback) {
-    var meta = {
-      relativePath: file,
-      filepath: path.resolve(file),
-      filename: path.basename(file),
-      basename: path.basename(file, path.extname(file)),
-      extension: path.extname(file),
-      parsed: false
+    if (results.__async) {
+      fs.lstat(results.meta.filepath, onLsStat);
+    } else {
+      onLsStat(null, fs.lstatSync(results.meta.filepath));
+    }
+  }
+  __getFileContents (done, results) {
```

Comment:
We should strive for consistency. Constructor gets like this: `constructor()` (no space after function name) but other functions do get the space. I think we've been using no space. 

@jgallen23 ?